index.bs

```diff
@@ -52,10 +69,175 @@ dictionary StorageBucketOptions {
 </xmp>
 
 <h3 id="storage-bucket-open">Creating a bucket</h3>
+
+<div algorithm>
+
+The <dfn method for="StorageBucketManager">open(|name|, |options|)</dfn> method steps are:
+
+1. Let |environment| be [=/this=]'s [=/relevant settings object=].
+    
+1. Let |shelf| be the result of running [=obtain a local storage shelf=] given |environment|.
+
+1. If |shelf| is failure, then [=exception/throw=] an "{{UnknownError}}" {{DOMException}} and abort these steps.
+
+1. Let |p| be [=a new promise=].
+
+1. Run the following steps [=in parallel=]:
+
+    1. If the result of [=validate a bucket name=] with |name| is failure, then [=/reject=] |p| with an {{TypeError}} and abort these steps.
+
+    1. Let |r| be the result of running [=open a bucket=] with |shelf|, |name|, and |options|.
+
+    1. If |r| is failure, then [=reject=] |p| with a {{TypeError}} and abort these steps.
+
+    1. [=/Resolve=] |p| with |r|.
+
+1. Return |p|.
+
+</div>
+
+<div algorithm>
+
+To <dfn>open a bucket</dfn> for a |shelf| given a bucket |name| and optional |options|, run the following steps:
+
+1. Let |expires| be undefined.
+    
+1. If |options|["{{StorageBucketOptions/expires}}"] exists, then:
+    
+    1. Let |expires| be the result of [=parse a duration string=] from |options|["{{StorageBucketOptions/expires}}"]
+    
+    1. If |expires| is not a [=duration=], then return failure.
+    
+    1. If |expires| is less than or equal to the [=current monotonic time=], then return failure.
+
+1. Let |quota| be |options|["{{StorageBucketOptions/quota}}"] if it exists, otherwise undefined.
+
+1. If |quota| is less than or equal to zero, then return failure.
+
+1. Let |persisted| be false.
+
+1. If |options|["{{StorageBucketOptions/persisted}}"] exists and is true, then:
+
+    1. Let |permission| be the result of [=/requesting permission to use=] `"persistent-storage"`.
+
+    1. If |permission| is "{{PermissionState/granted}}", then set |persisted| to true.
+
+1. Let |bucket| be |shelf|'s [=bucket map=][|name|] if exists or null otherwise.
+
+1. If |bucket| is non-null and |bucket|'s [=StorageBucket/expiration value=] is non-null and less than or equal the [=current monotonic time=], then:
+
+    1. Remove |bucket|.
```

Comment:
Bucket removal needs to be defined.  It probably makes sense to have a separate algorithm for this unless its as simple as removing from a map or something.  For example, the algorithm could remove from the bucket map, mark the bucket as "removed" so methods fail on it, etc.

Reply:
Added!